atom/browser/net/atom_network_delegate.cc

```diff
@@ -241,6 +243,12 @@ int AtomNetworkDelegate::OnBeforeSendHeaders(
     net::URLRequest* request,
     const net::CompletionCallback& callback,
     net::HttpRequestHeaders* headers) {
+  auto cmd_line = base::CommandLine::ForCurrentProcess();
+  auto client_id = cmd_line->GetSwitchValueASCII(
+      switches::kDevToolsEmulateNetworkConditionsClientId);
+  if (!client_id.empty())
+    headers->SetHeader(
+        switches::kDevToolsEmulateNetworkConditionsClientId, client_id);
```

Comment:
How does Chromium handle it? This change would add an extra header for every request, which does not look like a good thing.